app/src/org/commcare/tasks/ProcessAndSendTask.java

```diff
@@ -188,6 +188,7 @@ private boolean checkFormRecordStatus(FormRecord[] records)
                         record.logPendingDeletion(TAG,
                                 "the xml submission file associated with the record could not be found");
                         FormRecordCleanupTask.wipeRecord(c, record);
+                        records[i] = FormRecord.StandInForDeletedRecord();

```

Comment:
Question 2: What was happening here previously seems really shady to me, because the form gets wiped from the db, but then there is still a stale handle to that `FormRecord` object in the `records` array, which is what `sendForms()` goes on to use after this method call finishes. So it looks to me like we would have still been trying to send that stale record when we get to it in `sendForms()`, with unclear results. 

I'm proposing this as an alternate strategy, where we create a stand-in record with its own status type (`STATUS_JUST_DELETED`), and then check for this status explicitly in each iteration of `sendForms()` and consider the result for a record with this status to be `RECORD_FAILURE`

Reply:
Another option would be to have the `wipeRecord` method change the status of the original record object to `STATUS_JUST_DELETED`. No strong feelings here, agree that the previous behavior feels dangerous.

Reply:
Hm, I think I prefer this way because I think it's cleaner to represent the record with something separate, rather than to try to hang onto a stale pointer to a deleted record.

Reply:
StandIn approach makes sense to me. Agree that keeping the handle to the deleted record around feels quite dangerous if we truly think we should abandon it.